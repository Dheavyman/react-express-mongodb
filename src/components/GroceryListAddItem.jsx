import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  input: PropTypes.string.isRequired,
  handleChangeInputName: PropTypes.func.isRequired,
  addItem: PropTypes.func.isRequired
};

/**
 * Add grocery item component
 *
 * @param {string} input - grocery input
 * @param {function} handleChangeInputName - Change input name handler
 * @param {function} addItem - Add item handler
 *
 * @returns {object} React element
 */
const GroceryListAddItem = ({ input, handleChangeInputName, addItem }) => (
  <div className="col-sm pl-4">
    <form className="form-inline" onSubmit={addItem}>
      <div className="form-group mt-3 mt-sm-0">
        <label htmlFor="inputGroceryItem" className="sr-only">Grocery item</label>
        <input
          id="inputGroceryItem"
          type="text"
          className="form-control"
          placeholder="Grocery item"
          value={input}
          onChange={handleChangeInputName}
        />
      </div>
      <button type="submit" className="btn btn-outline-primary">
        Add Item
      </button>
    </form>
  </div>
);

GroceryListAddItem.propTypes = propTypes;

export default GroceryListAddItem;
