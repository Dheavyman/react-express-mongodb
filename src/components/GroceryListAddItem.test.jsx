import React from 'react';
import TestRenderer from 'react-test-renderer';

import GroceryListAddItem from './GroceryListAddItem';

describe('GroceryListAddItem component', () => {
  let props;
  beforeEach(() => {
    props = {
      input: 'Sugar',
      handleChangeInputName: jest.fn(),
      addItem: jest.fn()
    };
  });

  it('should render the component', () => {
    const renderer = TestRenderer.create(<GroceryListAddItem {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
  it('should call addItem handler to add a new grocery item', () => {
    const renderer = TestRenderer.create(<GroceryListAddItem {...props} />);

    renderer.root.findAllByType('form')[0].props.onSubmit(
      { preventDefault: jest.fn()});

    expect(props.addItem).toHaveBeenCalled();
  });
});
