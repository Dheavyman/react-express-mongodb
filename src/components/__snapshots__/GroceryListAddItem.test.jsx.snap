// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroceryListAddItem component should render the component 1`] = `
<div
  className="col-sm pl-4"
>
  <form
    className="form-inline"
    onSubmit={[MockFunction]}
  >
    <div
      className="form-group mt-3 mt-sm-0"
    >
      <label
        className="sr-only"
        htmlFor="inputGroceryItem"
      >
        Grocery item
      </label>
      <input
        className="form-control"
        id="inputGroceryItem"
        onChange={[MockFunction]}
        placeholder="Grocery item"
        type="text"
        value="Sugar"
      />
    </div>
    <button
      className="btn btn-outline-primary"
      type="submit"
    >
      Add Item
    </button>
  </form>
</div>
`;
